data_structures/trie/trie.py

```diff
@@ -46,6 +46,57 @@ def find(self, word: str) -> bool:
             curr = curr.nodes[char]
         return curr.is_leaf
 
+    def match(self, prefix: str) -> [str]:
+        """
+        gets the list of all the words that
+        are followed by certain prefix.
+    
+        long-url: https://www.geeksforgeeks.org/prefix-matching-python-using-pytrie-module/
+        :param prefix: the prefix to be matched
+        :return: List of prefix-matched words.
+
+        """
+        pass
+
+
+        def search(t: TrieNode, prefix: str) -> TrieNode:
```

Comment:
`search()` seems liek a weak function name.  What about `get_trie_for prefix()` or similar?

Doctests? How do we know this works?